src/main/scala/is/hail/table/Table.scala

```diff
@@ -402,29 +415,29 @@ class Table(val hc: HailContext, val tir: TableIR) {
 
   def head(n: Long): Table = new Table(hc, TableHead(tir, n))
 
-  def keyBy(key: String*): Table = keyBy(key.toArray, null)
+  def keyBy(key: String*): Table = keyBy(key.toFastIndexedSeq, null)
```

Comment:
I simplified the overload set a bit. `Table` still has an `Optional` key for now, so it's still helpful to have the overloads taking an `Optional` key (and it would be mildly annoying to have to write `Some` at most call sites). After table keys aren't optional, yes, there can be only two overloads.